src/core/include/mp-units/framework/representation_concepts.h

```diff
@@ -67,6 +70,26 @@ namespace detail {
 template<typename T>
 concept WeaklyRegular = std::copyable<T> && std::equality_comparable<T>;
 
+/**
+ * @brief MagnitudeScalable
+ *
+ * Physical quantities can be represented as a product of a "number" $n$ times a unit $u$.
+ * Because the units $u$ of each physical dimension form a (mathematical) vector space over
+ * the field of the real numbers, and each such unit admits an equivalent representation
+ * of the same physical quantitiy, the "number" will have to embed the structure of the
+ * same (mathematical) vector space over the real numbers. That is,
+ * for some scaled unit $u' = f \cdot u$ ($f \in \mathcal{R}$), the two representations
+ * $(n \cdot f) \times u$ and $n \times u'$ are equivalent, and thus the mathematical space
+ * embedding $n$ must admit scaling by a real number. Same for addition.
+ *
+ */
+
+template<typename T>
+concept MagnitudeScalable = detail::WeaklyRegular<T> && requires(T a, T b, std::type_identity<T> to_type) {
+  { mp_units::scale(mag<1>, a) };
```

Comment:
We could check return type here as well. For example, `WeaklyRegular` could be better than nothing here.